libmamba/src/specs/version.cpp

```diff
@@ -217,78 +216,108 @@ namespace mamba::specs
          * ``[1, 2, 0, 0]`` are considered equal, however ``[1, 2]`` and ``[1, 0, 2]`` are not.
          * Similarily ``[1, 1] is less than ``[1, 2, 0]`` but more than ``[1, 1, -1]``
          * because ``-1 < 0``.
+         *
+         * @return The comparison between the two sequence
+         * @return The first index where the two sequence diverge.
          */
-        template <typename Iter1, typename Iter2, typename T, typename Cmp>
+        template <typename Iter1, typename Iter2, typename Empty1, typename Empty2, typename Cmp>
         constexpr auto lexicographical_compare_three_way_trailing(
-            Iter1 first1,
-            Iter1 last1,
-            Iter2 first2,
-            Iter2 last2,
-            T empty,
+            const Iter1 first1,
+            const Iter1 last1,
+            const Iter2 first2,
+            const Iter2 last2,
+            const Empty1 empty1,
+            const Empty2 empty2,
```

Comment:
`const` are unnecessary for variables passed by value.

Reply:
They are not wrong either :) The previous version would `++first1` `++first2`, now with `iter1` and `iter2` there are six iterator with potentially the same type. I though I would be a good idea to make sure we only change the one we mean to.

Alternatively, the arguments were taken by value to imitate the stdlib but I wonder if `const&` would be better suited.

Reply:
Iterator arguments are usually passed by copy because they are usually modified inside the function, so this avoid to code an extra copy in it. Also, they are supposed to be lightweight object, so that the copy does not hurt performance.

> I though I would be a good idea to make sure we only change the one we mean to.

This is from the implementation point-of-view, from the API one, since they are passed by copy, the caller does not really care wht is changed and what is not.